### PR TITLE
added isoscelestriangle

### PR DESCRIPTION
**Pull request by kinoko-neko**

### CODE REVIEW

File: src/com/oneofever/shapes/IsoscelesTriangle.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+            area = values.get("area").b.get(0);
+
+            if (area < 0) {
+                throw new IllegalArgumentException("Area cannot be negative.");
+            }
+        }
+        if (values.containsKey("base")) {
+            base = values.get("base").b.get(0);
+
+            if (area < 0) {
```
Comment (Acors24) [inline src/com/oneofever/shapes/IsoscelesTriangle.java:40]: wrong condition?

File: src/com/oneofever/shapes/IsoscelesTriangle.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+            } else if (area != null) {
+                // from area formula
+                height = (2.0 * area) / base;
+                // from pithagoras theorem
+                side = Math.sqrt(height * height + ((base * base) / 4.0));
+            }
+        } else if (height != null) {
+            if (side != null) {
+                base = 2.0 * Math.sqrt(side * side - height * height);
+                area = area = (base * height) / 2.0;
```
Comment (Acors24) [inline src/com/oneofever/shapes/IsoscelesTriangle.java:63]: `area = area`
absolutely

File: src/com/oneofever/shapes/IsoscelesTriangle.java (hunk 1 above)
Comment (kinoko-neko) [reply src/com/oneofever/shapes/IsoscelesTriangle.java:40]: true

File: src/com/oneofever/shapes/IsoscelesTriangle.java (hunk 2 above)
Comment (kinoko-neko) [reply src/com/oneofever/shapes/IsoscelesTriangle.java:63]: if it works don't question it xd

**Review by hrmnpfg — COMMENTED** [review]
image
almost working?

**Review by Acors24 — APPROVED** [review]

**Pull request merged**